test/test_unistd.rs

```diff
@@ -291,3 +292,26 @@ fn test_sysconf_unsupported() {
     let open_max = sysconf(SysconfVar::_XOPEN_CRYPT);
     assert!(open_max.expect("sysconf failed").is_none())
 }
+
+// Test that we can create a pair of pipes.  No need to verify that they pass
+// data; that's the domain of the OS, not nix.
+#[test]
+fn test_pipe() {
+    let (fd0, fd1) = pipe().unwrap();
+    let m0 = stat::SFlag::from_bits_truncate(stat::fstat(fd0).unwrap().st_mode);
+    // S_IFIFO means it's a pipe
+    assert_eq!(m0, stat::S_IFIFO);
+    let m1 = stat::SFlag::from_bits_truncate(stat::fstat(fd1).unwrap().st_mode);
+    assert_eq!(m1, stat::S_IFIFO);
+}
+
+// pipe2(2) is the same as pipe(2), except it allows setting some flags.  Check
+// that we can set a flag.
+#[test]
+fn test_pipe2() {
+    let (fd0, fd1) = pipe2(fcntl::O_CLOEXEC).unwrap();
+    let f0 = fcntl::FdFlag::from_bits_truncate(fcntl::fcntl(fd0, fcntl::F_GETFD).unwrap());
+    assert!(f0.contains(fcntl::FD_CLOEXEC));
```

Comment:
Shouldn't you check that both of these `fd`s don't have this flag to begin with?

Reply:
What do you mean "to begin with"?  `pipe2` sets that flag atomically as the pipe is created.

Reply:
I ask because in the case that `FD_CLOEXEC` is "default", this test could still pass even though the desired behavior didn't necessarily happen. I wanted to confirm that this assumption is correct.

Reply:
I ask because in the case that `FD_CLOEXEC` is "default", this test could still pass even though the desired behavior didn't necessarily happen. I wanted to confirm that this assumption is correct.

Reply:
You're correct, ignore this comment.